src/attributedlabel/src/NIAttributedLabel.m

```diff
@@ -214,7 +214,7 @@ @interface NIAttributedLabel() <UIActionSheetDelegate>
 
 @property (nonatomic) CTFrameRef textFrame; // CFType, manually managed lifetime, see setter.
 
-@property (assign)            BOOL detectingLinks; // Atomic.
+@property (nonatomic, assign) NSInteger linkDetectionGeneration;
```

Comment:
I think semaphore has a very specific meaning  relating to synchronization primitives that would be a bit misleading since this isn't doing any locking. Maybe the "generation" naming is a bit vague too. How about currentLinkDetectionRequest, or something similar?